corehq/apps/locations/models.py

```diff
@@ -55,6 +58,14 @@ def root_locations(cls, domain, include_archive_ancestors=False):
         roots = cls.objects.root_nodes().filter(domain=domain)
         return _filter_for_archived(roots, include_archive_ancestors)
 
+    @memoized
+    def get_products(self):
+        """
+        If there are no products specified for this location, assume all
+        products for the domain are relevant.
+        """
+        return self.products.all() or SQLProduct.by_domain(self.domain)
```

Comment:
I guess I don't know what the UI looks like, but could you uncheck all products and then magically have them all show up again?